migration/init.go

```diff
@@ -16,7 +16,42 @@ var _ weave.Initializer = Initializer{}
 // and save it to the database
 func (Initializer) FromGenesis(opts weave.Options, kv weave.KVStore) error {
 	if err := gconf.InitConfig(kv, opts, "migration", &Configuration{}); err != nil {
-		return errors.Wrap(err, "init config")
+		return errors.Wrap(err, "migration config")
 	}
+
+	var packages []string
+	if err := opts.ReadOptions("initialize_schema", &packages); err != nil {
+		return errors.Wrap(err, "initialize schema")
+	}
+
+	b := NewSchemaBucket()
+
+	// Before ensuring the schema of above packages is initialized force
+	// register migration package schema.
+	// This is solving a chicken-egg problem. We could not register any
+	// schema version without Schema model being enabled (schema registered
+	// with version one).
+	_, err := b.Create(kv, &Schema{
+		Metadata: &weave.Metadata{Schema: 1},
+		Pkg:      "migration",
+		Version:  1,
+	})
+	// Duplicated initializations are ignored.
+	if err != nil && !errors.ErrDuplicate.Is(err) {
+		return errors.Wrap(err, "initialize migration package schema")
+	}
+
+	for _, name := range packages {
+		_, err := b.Create(kv, &Schema{
```

Comment:
As mentioned above, this hardcodes `Version: 1`. I can imagine in the future, some modules will be at 2 or 3 from genesis, as we start with the newest code.

Can you change `packages` from `[]string` to
```go
[]struct{
  Pkg string `json:"pkg"`
  Version int `json:"version"`
}
```
And default Version to 1 if it is 0.

So, we can have a simple but flexible init like:

```golang
"initialize_schema": []SchemaInit{
			{Pkg: "batch"},
			{Pkg: "cash", Version: 2},
                        // ...
```

Reply:
Done in https://github.com/iov-one/weave/pull/556/commits/3be47627e93a5c446b22bd4264c029394ef1f653